internal/api/auth/basic_auth.go

```diff
@@ -12,6 +13,44 @@ const (
 	realm                = `Basic realm="restricted"`
 )
 
+type authMode int
+
+const (
+	NoAuth authMode = iota
```

Comment:
Is it OK to make the `NoAuth` method default ?

Reply:
Yes for now.
This only protects OpenConfig data, one could argue this is critical, one could argue it is not.
In a near future, we will basic authentication without LDAP (or using TOKEN), which will be the default.